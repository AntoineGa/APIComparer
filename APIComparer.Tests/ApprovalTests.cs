using System;
using System.IO;
using APIComparer.Outputters;
using ApprovalTests;
using NuGet;
using NUnit.Framework;

namespace APIComparer.Tests
{
    [TestFixture]
    public class ApprovalTests
    {
        [Test]
        public void ApproveMarkdown()
        {
            var packages = new[] { "4.6.3", "5.0.0-beta0002" };

            var repo = new AggregateRepository(new[] {
                PackageRepositoryFactory.Default.CreateRepository("https://www.nuget.org/api/v2"),
                PackageRepositoryFactory.Default.CreateRepository(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "NuGet", "Cache"))
            });

            var packageManager = new PackageManager(repo, "packages");

            foreach (var v in packages)
            {
                if (!Directory.Exists(Path.Combine("packages", "NServiceBus." + v)))
                    packageManager.InstallPackage("NServiceBus", SemanticVersion.Parse(v));
            }

            var file1 = Path.Combine("packages", "NServiceBus." + packages[0], "lib", "net40", "NServiceBus.Core.dll");
            var file2 = Path.Combine("packages", "NServiceBus." + packages[1], "lib", "net45", "NServiceBus.Core.dll");

            var engine = new ComparerEngine();
            engine.Filter = new NServiceBusAPIFilter();

            var diff = engine.CreateDiff(file1, file2);

            var formatter = new APIUpgradeToMarkdownFormatter("Result.md");

            formatter.WriteOut(diff);

            Approvals.Verify(File.ReadAllText("Result.md"));
        }
    }
}