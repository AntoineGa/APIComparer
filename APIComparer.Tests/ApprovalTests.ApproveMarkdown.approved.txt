The following types are missing in the new API.

    NServiceBus.Audit.MessageAuditer
    NServiceBus.AutomaticSubscriptions.AutoSubscriber
    NServiceBus.AutomaticSubscriptions.DefaultAutoSubscriptionStrategy
    NServiceBus.AutomaticSubscriptions.IAutoSubscriptionStrategy
    NServiceBus.CircuitBreakers.CircuitBreaker
    NServiceBus.Config.AddressInitializer
    NServiceBus.Config.ChannelCollection
    NServiceBus.Config.ChannelConfig
    NServiceBus.Config.DisplayInfrastructureServicesStatus
    NServiceBus.Config.GatewayConfig
    NServiceBus.Config.InfrastructureServices
    NServiceBus.Config.SatelliteConfigurer
    NServiceBus.Config.SiteCollection
    NServiceBus.Config.SiteConfig
    NServiceBus.Config.WindowsInstallerRunner
    NServiceBus.ConfigureGateway
    NServiceBus.ConfigureSagas
    NServiceBus.ConfigureSecondLevelRetriesExtensions
    NServiceBus.DataBus.Config.Bootstrapper
    NServiceBus.DataBus.DefaultDataBusSerializer
    NServiceBus.DataBus.FileShare.FileShareDataBus
    NServiceBus.DataBus.InMemory.InMemoryDataBus
    NServiceBus.Encryption.EncryptionMessageMutator
    NServiceBus.Faults.Forwarder.Config.FaultsQueueCreator
    NServiceBus.Faults.Forwarder.FaultManager
    NServiceBus.Faults.InMemory.FaultManager
    NServiceBus.Features.Categories.Serializers
    NServiceBus.Features.EnableDefaultFeatures
    NServiceBus.Features.Feature`1
    NServiceBus.Features.FeatureCategory
    NServiceBus.Features.FeatureInitializer
    NServiceBus.Features.Gateway
    NServiceBus.FeatureSettingsExtensions
    NServiceBus.Gateway.Channels.Channel
    NServiceBus.Gateway.Channels.ChannelFactory
    NServiceBus.Gateway.Channels.ChannelTypeAttribute
    NServiceBus.Gateway.Channels.DataReceivedOnChannelArgs
    NServiceBus.Gateway.Channels.Http.DefaultResponder
    NServiceBus.Gateway.Channels.Http.HttpChannelReceiver
    NServiceBus.Gateway.Channels.Http.HttpChannelSender
    NServiceBus.Gateway.Channels.Http.HttpHeaders
    NServiceBus.Gateway.Channels.Http.IHttpResponder
    NServiceBus.Gateway.Channels.Http.SetDefaultResponder
    NServiceBus.Gateway.Channels.IChannelFactory
    NServiceBus.Gateway.Channels.IChannelReceiver
    NServiceBus.Gateway.Channels.IChannelSender
    NServiceBus.Gateway.Channels.ReceiveChannel
    NServiceBus.Gateway.Deduplication.GatewayMessage
    NServiceBus.Gateway.Deduplication.InMemoryDeduplication
    NServiceBus.Gateway.Deduplication.RavenDBDeduplication
    NServiceBus.Gateway.DefaultInputAddress
    NServiceBus.Gateway.HeaderManagement.DataBusHeaderManager
    NServiceBus.Gateway.HeaderManagement.GatewayHeaderManager
    NServiceBus.Gateway.HeaderManagement.GatewayHeaders
    NServiceBus.Gateway.HeaderManagement.HeaderMapper
    NServiceBus.Gateway.Notifications.IMessageNotifier
    NServiceBus.Gateway.Notifications.INotifyAboutMessages
    NServiceBus.Gateway.Notifications.MessageNotifier
    NServiceBus.Gateway.Notifications.MessageReceivedOnChannelArgs
    NServiceBus.Gateway.Persistence.InMemoryPersistence
    NServiceBus.Gateway.Persistence.MessageInfo
    NServiceBus.Gateway.Persistence.Raven.GatewayMessage
    NServiceBus.Gateway.Persistence.Raven.RavenDbPersistence
    NServiceBus.Gateway.Receiving.ChannelException
    NServiceBus.Gateway.Receiving.ConfigurationBasedChannelManager
    NServiceBus.Gateway.Receiving.ConventionBasedChannelManager
    NServiceBus.Gateway.Receiving.GatewayReceiver
    NServiceBus.Gateway.Receiving.IManageReceiveChannels
    NServiceBus.Gateway.Receiving.IReceiveMessagesFromSites
    NServiceBus.Gateway.Receiving.SingleCallChannelReceiver
    NServiceBus.Gateway.Routing.Endpoints.DefaultEndpointRouter
    NServiceBus.Gateway.Routing.IRouteMessagesToEndpoints
    NServiceBus.Gateway.Routing.IRouteMessagesToSites
    NServiceBus.Gateway.Routing.Site
    NServiceBus.Gateway.Routing.Sites.ConfigurationBasedSiteRouter
    NServiceBus.Gateway.Routing.Sites.KeyPrefixConventionSiteRouter
    NServiceBus.Gateway.Routing.Sites.OriginatingSiteHeaderRouter
    NServiceBus.Gateway.Sending.CallInfo
    NServiceBus.Gateway.Sending.CallType
    NServiceBus.Gateway.Sending.GatewaySender
    NServiceBus.Gateway.Sending.IForwardMessagesToSites
    NServiceBus.Gateway.Sending.SingleCallChannelForwarder
    NServiceBus.Gateway.Utils.Hasher
    NServiceBus.Hosting.Configuration.ConfigManager
    NServiceBus.Hosting.Wcf.WcfServiceHost
    NServiceBus.Impersonation.Windows.WindowsIdentityEnricher
    NServiceBus.Installation.GatewayHttpListenerInstaller
    NServiceBus.Installation.PerformanceMonitorUsersInstaller
    NServiceBus.Management.Retries.SecondLevelRetries
    NServiceBus.ObjectBuilder.Common.CommonObjectBuilder
    NServiceBus.ObjectBuilder.Common.Config.ConfigureCommon
    NServiceBus.ObjectBuilder.Common.SynchronizedInvoker
    NServiceBus.Persistence.InMemory.InMemoryPersistence
    NServiceBus.Persistence.InMemory.SagaPersister.InMemorySagaPersister
    NServiceBus.Persistence.InMemory.SubscriptionStorage.InMemorySubscriptionStorage
    NServiceBus.Persistence.InMemory.TimeoutPersister.InMemoryTimeoutPersistence
    NServiceBus.Persistence.Msmq.SubscriptionStorage.Config.SubscriptionsQueueCreator
    NServiceBus.Persistence.Msmq.SubscriptionStorage.Entry
    NServiceBus.Persistence.Msmq.SubscriptionStorage.MsmqSubscriptionStorage
    NServiceBus.Persistence.Raven.RavenConventions
    NServiceBus.Persistence.Raven.RavenPersistenceConstants
    NServiceBus.Persistence.Raven.RavenSessionFactory
    NServiceBus.Persistence.Raven.RavenUnitOfWork
    NServiceBus.Persistence.Raven.RavenUserInstaller
    NServiceBus.Persistence.Raven.SagaPersister.RavenSagaPersister
    NServiceBus.Persistence.Raven.SagaPersister.SagaUniqueIdentity
    NServiceBus.Persistence.Raven.StoreAccessor
    NServiceBus.Persistence.Raven.SubscriptionStorage.MessageTypeConverter
    NServiceBus.Persistence.Raven.SubscriptionStorage.RavenSubscriptionStorage
    NServiceBus.Persistence.Raven.SubscriptionStorage.Subscription
    NServiceBus.Persistence.Raven.TimeoutPersister.RavenTimeoutPersistence
    NServiceBus.Persistence.SetupDefaultPersistence
    NServiceBus.Sagas.ConfigureHowToFindSagaWithMessageDispatcher
    NServiceBus.Sagas.ConfigureTimeoutAsSystemMessages
    NServiceBus.Sagas.Finders.HeaderSagaIdFinder`1
    NServiceBus.Sagas.Finders.PropertySagaFinder`2
    NServiceBus.Satellites.Config.SatelliteContext
    NServiceBus.Satellites.SatelliteLauncher
    NServiceBus.Satellites.SatellitesQueuesCreator
    NServiceBus.Scheduling.Configuration.ConfigureScheduledTaskAsSystemMessages
    NServiceBus.Scheduling.Configuration.SchedulerConfiguration
    NServiceBus.Scheduling.DefaultScheduler
    NServiceBus.Scheduling.InMemoryScheduledTaskStorage
    NServiceBus.Scheduling.IScheduledTaskStorage
    NServiceBus.Scheduling.IScheduler
    NServiceBus.Scheduling.Messages.ScheduledTask
    NServiceBus.Scheduling.ScheduledTaskMessageHandler
    NServiceBus.SecondLevelRetries.Helpers.SecondLevelRetriesHeaders
    NServiceBus.SecondLevelRetries.Helpers.TransportMessageHelpers
    NServiceBus.SecondLevelRetries.SecondLevelRetriesProcessor
    NServiceBus.Serializers.Json.Internal.MessageContractResolver
    NServiceBus.Serializers.Json.Internal.MessageSerializationBinder
    NServiceBus.Serializers.Json.Internal.XContainerConverter
    NServiceBus.Serializers.XML.Config.MessageTypesInitializer
    NServiceBus.Serializers.XML.XmlSanitizingStream
    NServiceBus.ServiceAsyncResult
    NServiceBus.Settings.Endpoint
    NServiceBus.Settings.ISetDefaultSettings
    NServiceBus.Settings.TransportSettings
    NServiceBus.SyncConfig
    NServiceBus.Timeout.Core.DefaultTimeoutManager
    NServiceBus.Timeout.Core.TimeoutManagerDefaults
    NServiceBus.Timeout.Hosting.Windows.TimeoutDispatcherProcessor
    NServiceBus.Timeout.Hosting.Windows.TimeoutMessageProcessor
    NServiceBus.Timeout.Hosting.Windows.TimeoutPersisterReceiver
    NServiceBus.Timeout.TimeoutManagerDeferrer
    NServiceBus.Timeout.TimeoutManagerHeaders
    NServiceBus.Transports.Msmq.Config.CheckMachineNameForComplianceWithDtcLimitation
    NServiceBus.Transports.Msmq.MsmqDequeueStrategy
    NServiceBus.Transports.Msmq.MsmqMessageSender
    NServiceBus.Transports.Msmq.MsmqQueueCreator
    NServiceBus.Unicast.BusAsyncResultEventArgs
    NServiceBus.Unicast.Config.DefaultToTimeoutManagerBasedDeferal
    NServiceBus.Unicast.Config.DefaultTransportForHost
    NServiceBus.Unicast.DefaultDispatcherFactory
    NServiceBus.Unicast.IMessageDispatcherFactory
    NServiceBus.Unicast.MessagingBestPractices
    NServiceBus.Unicast.Monitoring.CausationMutator
    NServiceBus.Unicast.Monitoring.CriticalTimeCalculator
    NServiceBus.Unicast.Monitoring.EstimatedTimeToSLABreachCalculator
    NServiceBus.Unicast.Monitoring.PerformanceCounterInitializer
    NServiceBus.Unicast.Monitoring.ProcessingStatistics
    NServiceBus.Unicast.Publishing.StorageDrivenPublisher
    NServiceBus.Unicast.Queuing.Installers.AuditQueueCreator
    NServiceBus.Unicast.Queuing.Installers.EndpointInputQueueCreator
    NServiceBus.Unicast.Queuing.Installers.ForwardReceivedMessagesToQueueCreator
    NServiceBus.Unicast.Queuing.QueuesCreator
    NServiceBus.Unicast.Subscriptions.MessageDrivenSubscriptions.EnableMessageDrivenPublisherIfStorageIsFound
    NServiceBus.Unicast.Subscriptions.MessageDrivenSubscriptions.MessageDrivenSubscriptionManager
    NServiceBus.Unicast.Subscriptions.MessageDrivenSubscriptions.NoopSubscriptionAuthorizer
    NServiceBus.Unicast.Subscriptions.MessageDrivenSubscriptions.SubcriberSideFiltering.FilteringMutator
    NServiceBus.Unicast.Subscriptions.MessageDrivenSubscriptions.SubcriberSideFiltering.SubscriptionPredicatesEvaluator
    NServiceBus.Unicast.Transport.Config.Bootstrapper
    NServiceBus.Unicast.Transport.Transactional.Config.AdvancedTransactionalConfig
    NServiceBus.Unicast.Transport.TransportConnectionString
    NServiceBus.Unicast.Transport.TransportMessageExtensions

The following members are missing on the public types.

    NServiceBus.Config.SatelliteConfigurer
        System.Void NServiceBus.Config.SatelliteConfigurer::Init()

    NServiceBus.Faults.Forwarder.Config.FaultsQueueCreator
        System.Boolean NServiceBus.Faults.Forwarder.Config.FaultsQueueCreator::get_IsDisabled()

    NServiceBus.Installation.PerformanceMonitorUsersInstaller
        System.Void NServiceBus.Installation.PerformanceMonitorUsersInstaller::Install(System.String)

    NServiceBus.ObjectBuilder.Common.CommonObjectBuilder
        NServiceBus.ObjectBuilder.IComponentConfig NServiceBus.ObjectBuilder.Common.CommonObjectBuilder::ConfigureComponent(System.Type,NServiceBus.ObjectBuilder.ComponentCallModelEnum)
        NServiceBus.ObjectBuilder.IComponentConfig`1<T> NServiceBus.ObjectBuilder.Common.CommonObjectBuilder::ConfigureComponent(NServiceBus.ObjectBuilder.ComponentCallModelEnum)
        System.Void NServiceBus.ObjectBuilder.Common.CommonObjectBuilder::DisposeManaged()

    NServiceBus.Sagas.Finders.PropertySagaFinder`2
        System.Reflection.PropertyInfo NServiceBus.Sagas.Finders.PropertySagaFinder`2::get_MessageProperty()
        System.Reflection.PropertyInfo NServiceBus.Sagas.Finders.PropertySagaFinder`2::get_SagaProperty()
        System.Void NServiceBus.Sagas.Finders.PropertySagaFinder`2::set_MessageProperty(System.Reflection.PropertyInfo)
        System.Void NServiceBus.Sagas.Finders.PropertySagaFinder`2::set_SagaProperty(System.Reflection.PropertyInfo)
        TSaga NServiceBus.Sagas.Finders.PropertySagaFinder`2::FindBy(TMessage)

    NServiceBus.Satellites.SatelliteLauncher
        NServiceBus.ObjectBuilder.IBuilder NServiceBus.Satellites.SatelliteLauncher::get_Builder()
        System.Void NServiceBus.Satellites.SatelliteLauncher::set_Builder(NServiceBus.ObjectBuilder.IBuilder)

    NServiceBus.Satellites.SatellitesQueuesCreator
        System.Void NServiceBus.Satellites.SatellitesQueuesCreator::Install(System.String)

    NServiceBus.Scheduling.DefaultScheduler
        System.Void NServiceBus.Scheduling.DefaultScheduler::Schedule(NServiceBus.Scheduling.ScheduledTask)

    NServiceBus.Serializers.XML.Config.MessageTypesInitializer
        System.Void NServiceBus.Serializers.XML.Config.MessageTypesInitializer::Run()

    NServiceBus.Timeout.Core.DefaultTimeoutManager
        System.Void NServiceBus.Timeout.Core.DefaultTimeoutManager::add_TimeoutPushed(System.EventHandler`1<NServiceBus.Timeout.Core.TimeoutData>)
        System.Void NServiceBus.Timeout.Core.DefaultTimeoutManager::remove_TimeoutPushed(System.EventHandler`1<NServiceBus.Timeout.Core.TimeoutData>)

    NServiceBus.Timeout.Hosting.Windows.TimeoutMessageProcessor
        NServiceBus.Timeout.Core.IManageTimeouts NServiceBus.Timeout.Hosting.Windows.TimeoutMessageProcessor::get_TimeoutManager()
        System.Void NServiceBus.Timeout.Hosting.Windows.TimeoutMessageProcessor::set_TimeoutManager(NServiceBus.Timeout.Core.IManageTimeouts)

    NServiceBus.Timeout.Hosting.Windows.TimeoutPersisterReceiver
        NServiceBus.Timeout.Core.IManageTimeouts NServiceBus.Timeout.Hosting.Windows.TimeoutPersisterReceiver::get_TimeoutManager()
        System.Void NServiceBus.Timeout.Hosting.Windows.TimeoutPersisterReceiver::set_TimeoutManager(NServiceBus.Timeout.Core.IManageTimeouts)

    NServiceBus.Timeout.TimeoutManagerDeferrer
        System.Void NServiceBus.Timeout.TimeoutManagerDeferrer::Defer(NServiceBus.TransportMessage,System.DateTime,NServiceBus.Address)

    NServiceBus.Transports.Msmq.MsmqMessageSender
        System.Void NServiceBus.Transports.Msmq.MsmqMessageSender::Send(NServiceBus.TransportMessage,NServiceBus.Address)

    NServiceBus.Unicast.MessagingBestPractices
        System.Void NServiceBus.Unicast.MessagingBestPractices::AssertIsValidForPubSub(System.Type)
        System.Void NServiceBus.Unicast.MessagingBestPractices::AssertIsValidForReply(System.Collections.Generic.IEnumerable`1<System.Object>)
        System.Void NServiceBus.Unicast.MessagingBestPractices::AssertIsValidForSend(System.Type,NServiceBus.MessageIntentEnum)

    NServiceBus.Unicast.Monitoring.CausationMutator
        System.Void NServiceBus.Unicast.Monitoring.CausationMutator::Init()
        System.Void NServiceBus.Unicast.Monitoring.CausationMutator::MutateOutgoing(System.Object[],NServiceBus.TransportMessage)

    NServiceBus.Unicast.Monitoring.CriticalTimeCalculator
        System.Void NServiceBus.Unicast.Monitoring.CriticalTimeCalculator::DisposeManaged()

    NServiceBus.Unicast.Monitoring.EstimatedTimeToSLABreachCalculator
        System.Void NServiceBus.Unicast.Monitoring.EstimatedTimeToSLABreachCalculator::DisposeManaged()

    NServiceBus.Unicast.Monitoring.PerformanceCounterInitializer
        System.Void NServiceBus.Unicast.Monitoring.PerformanceCounterInitializer::Run()

    NServiceBus.Unicast.Monitoring.ProcessingStatistics
        System.Void NServiceBus.Unicast.Monitoring.ProcessingStatistics::Init()

    NServiceBus.Unicast.Publishing.StorageDrivenPublisher
        System.Boolean NServiceBus.Unicast.Publishing.StorageDrivenPublisher::Publish(NServiceBus.TransportMessage,System.Collections.Generic.IEnumerable`1<System.Type>)

    NServiceBus.Unicast.Queuing.Installers.AuditQueueCreator
        NServiceBus.Audit.MessageAuditer NServiceBus.Unicast.Queuing.Installers.AuditQueueCreator::get_Auditer()
        System.Boolean NServiceBus.Unicast.Queuing.Installers.AuditQueueCreator::get_IsDisabled()
        System.Void NServiceBus.Unicast.Queuing.Installers.AuditQueueCreator::set_Auditer(NServiceBus.Audit.MessageAuditer)

    NServiceBus.Unicast.Queuing.Installers.EndpointInputQueueCreator
        System.Boolean NServiceBus.Unicast.Queuing.Installers.EndpointInputQueueCreator::get_IsDisabled()

    NServiceBus.Unicast.Queuing.Installers.ForwardReceivedMessagesToQueueCreator
        System.Boolean NServiceBus.Unicast.Queuing.Installers.ForwardReceivedMessagesToQueueCreator::get_IsDisabled()

    NServiceBus.Unicast.Queuing.QueuesCreator
        System.Void NServiceBus.Unicast.Queuing.QueuesCreator::Init()
        System.Void NServiceBus.Unicast.Queuing.QueuesCreator::Install(System.String)

    NServiceBus.Unicast.Transport.TransportConnectionString
        System.String NServiceBus.Unicast.Transport.TransportConnectionString::DefaultConnectionStringName
        System.String NServiceBus.Unicast.Transport.TransportConnectionString::GetConnectionStringOrNull(System.String)
        System.Void NServiceBus.Unicast.Transport.TransportConnectionString::Override(System.Func`1<System.String>)

